src/vm/assembly.cpp

```diff
@@ -2588,6 +2588,7 @@ HRESULT RunMain(MethodDesc *pFD ,
             // Set the return value to 0 instead of returning random junk
             *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal = GetLatchedExitCode();
```

Comment:
This line should not be executing at all for the test you are trying to fix. `System.Environment::Exit` should exit the process immediately, before the main method returns.